### PR TITLE
Adding IO rules to set PF cluster isolation in Ele/Phos

### PR DESCRIPTION
**Pull request by Sam-Harper**
As reported by @aminnj  at https://github.com/cms-sw/cmssw/issues/25573, in 102X its not possible to read the pf cluster isolation in pat::Electrons/Photons for 94X (or earlier objects). This is because in 101X we moved the pf cluster isolation from the pat::Electron/pat::Photon to the reco::GsfElectron/reco::Photon.

This PR adds an IO rule which fixes this and now allows PF cluster isolation of 94X produced objects to be read in 10X. 

Note, I couldnt figure out a better way to do this, there wasnt a good example of how to fill a base class member variable from a derived class member variable so I stuck with this simple solution, improvements are welcome. 

Obviously this only works when reading pat::Electrons/pat::Photons, reco:GsfElectrons/reco::Photons do not have the member in 94X and therefore it cant be filled.

A backport is required for 102X which I'll submit when this is mostly done.

### CODE REVIEW

File: DataFormats/PatCandidates/src/classes_def_objects.xml [hunk 1]
```diff
@@ -234,6 +237,9 @@
   <ioread sourceClass="pat::Photon" targetClass="pat::Photon" version="[1-]" source="" target="superClusterRelinked_">
     <![CDATA[superClusterRelinked_.reset();]]>
   </ioread>
+  <ioread sourceClass="pat::Photon"  targetClass="pat::Photon" version="[-20]" target="" source="float ecalPFClusIso_;float hcalPFClusIso_">
+    <![CDATA[reco::Photon::PflowIsolationVariables pfIsoVar = newObj->getPflowIsolationVariables();pfIsoVar.sumEcalClusterEt = onfile.ecalPFClusIso_;pfIsoVar.sumHcalClusterEt = onfile.hcalPFClusIso_;newObj->setPflowIsolationVariables(pfIsoVar);]]>
```
Comment (slava77) [inline DataFormats/PatCandidates/src/classes_def_objects.xml]: here and above: some newlines would be nice.

**Comment by slava77** [comment]
@cmsbuild please test

File: DataFormats/PatCandidates/src/classes_def_objects.xml (hunk 1 above)
Comment (Sam-Harper) [reply DataFormats/PatCandidates/src/classes_def_objects.xml]: done

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by perrotta** [comment]
+1
- Tested to fix the issue reported in https://github.com/cms-sw/cmssw/issues/25573
- Jenkins tests pass

**Comment by fabiocos** [comment]
code-checks

**Comment by fabiocos** [comment]
+1

**Comment by fabiocos** [comment]
merge

@Sam-Harper thank you, this is indeed useful to help in moving forward analysis code

**Pull request merged**